File: unit-3/sprint-1/day-5_Fragment Communication/we/Between Two Fragment/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.widget.FrameLayout;
public class MainActivity extends AppCompatActivity implements CommunicationListener {

    private FrameLayout frameLayout;
    private FragmentManager fragmentManager;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        frameLayout = findViewById(R.id.flContainer);
        launchStudentPersonalDetailsFrag();
    }

    private void launchStudentPersonalDetailsFrag() {
        fragmentManager = getSupportFragmentManager();
        StudentPersonalDetailsFragment studentPersonalDetailsFragment = new StudentPersonalDetailsFragment();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.flContainer, studentPersonalDetailsFragment, "StudentPersonal").commit();
    }

    @Override
    public void launchStudentPerformanceDetailsFrag(Bundle bundle) {
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        StudentPerformanceDetailsFragment studentPerformanceDetailsFragment = new StudentPerformanceDetailsFragment();
        studentPerformanceDetailsFragment.setArguments(bundle);
        fragmentTransaction.replace(R.id.flContainer,studentPerformanceDetailsFragment,"StudentPerformance").addToBackStack("StudentPerformance").commit();
    }
}

File: unit-5/data binding/Test/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.net.Network
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.myapplication.Network.Companion.getRetrofitInstance
import com.example.myapplication.databinding.ActivityMainBinding
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    lateinit var activityMainBinding: ActivityMainBinding
    lateinit var popularMoviesAdapter: PopularMoviesAdapter

    private var responseList = listOf<Result>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        callAPI()
        setRecycler()
    }
    private fun setRecycler() {
        activityMainBinding.apply {
            popularMoviesAdapter = PopularMoviesAdapter(responseList)
            rvPopularMovies.layoutManager = LinearLayoutManager(this@MainActivity)
            rvPopularMovies.adapter = popularMoviesAdapter
        }
    }

    private fun callAPI() {
        val apiClient = Network.getRetrofitInstance.create(APIClient::class.java)
            apiClient.getAPIResponse().enqueue(object : Callback<List<Result>> {
                override fun onResponse(call: Call<List<Result>>, response: Response<List<Result>>) {
                    responseList = response.body()!!
                }
                override fun onFailure(call: Call<List<Result>>, t: Throwable) {

                }
            })
    }
}